content/docs/1.0.0/tutorial/create-your-first-microservice-javascript.md

```diff
@@ -76,97 +88,91 @@ const iofog = require('@iofog/nodejs-sdk');
+For the curious, the first argument is the host name of the [Agent's Local API](../agents/local-api.html), the second is the port number of the SDK, and the third can be the container's ID, though it is not required.
 
 #### iofog.getConfig()
 
-Asynchronously fetch the microservice's current configuration (config).
+Asynchronously fetch the microservice's current configuration (config). The configuration of the microservice is only read once when it starts.
 
 ```js
-iofog.getConfig({
-  onNewConfig: nextConfig => {
-    // Do something with config
-    console.log(nextConfig);
-  },
-  onBadRequest: err => console.error('getConfig failed: ', err),
-  onError: err => console.error('getConfig failed: ', err)
-});
+function updateConfig() {
+    iofog.getConfig({
+        onNewConfig: newConfig => {
+            config = newConfig;
+        },
+        onBadRequest: err => console.error('updateConfig failed: ', err),
+        onError: err => console.error('updateConfig failed: ', err)
+    });
+}
 ```
 
 #### iofog.wsControlConnection()
 
-Connect to the ioFog Controller signal channel via WebSocket, which is used to receive notifications from the Controller that our microservice's config has changed.
+Note that when a configuration of a microservice changes, the Controller will send a message to the involved microservice.
 
-Because a config can be any arbitrary JSON, including very large files, the change notifications themselves do not actually include the config. So if you do in fact want to update your local cache of the config, you can follow up a change notification with a call to `iofog.getConfig()`.
+Therefore we have to connect the ioFog *control channel* via WebSocket, which is used to receive notifications from the Controller that our microservice's config has changed.
+
+Because a config can be any arbitrary JSON, including very large files, the change notifications themselves do not actually include the config. So if you do in fact want to update your local cache of the config, you have to follow up a change notification with a call to `iofog.getConfig()`.
 
 ```js
 iofog.wsControlConnection({
-  onNewConfigSignal: () => {
-    // notification-only, if you want to the updated config
-    // you'll need to subsequently use iofog.getConfig()
-    console.log('config has changed');
-  },
-  onError: err => console.error('Error with Control Connection: ', err)
+    onNewConfigSignal: () => updateConfig(),
+    onError: err => console.error('Error with Control Connection: ', err)
 });
 ```
 
 #### iofog.wsMessageConnection()
 
-Connect to the ioFog message channel via WebSocket. This is where you'll receive any messages routed to this microservice from another.
+Next, we have to connect to the ioFog *message channel* via WebSocket. This is where you'll receive any messages routed to this microservice from another.
```

Comment:
_Edits are bolded._

This is where **we'll** receive any messages routed to this microservice from another.